analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count.Rmd

```diff
@@ -0,0 +1,375 @@
+
+# Set up
+```{r load-library}
+suppressPackageStartupMessages({
+  library(tidyverse)
+  library(flextable)
+})
+```
+
+## Directories and File Inputs/Outputs
+```{r set-dir-and-file-names}
+# Detect the ".git" folder -- this will be in the project root directory.
+# Use this as the root directory to ensure proper sourcing of functions no
+# matter where this is called from
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+analysis_dir <- file.path(root_dir, "analyses", "sample-distribution-analysis")
+input_dir <- file.path(analysis_dir, "input")
+
+# Inputs
+v12_file <- file.path(data_dir, "histologies.tsv")
+
+# File path to results directory
+results_dir <-
+  file.path(analysis_dir, "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+# Read in data and process Data
+```{r cohort PBTA, echo=TRUE}
+#---------------------------------------------------------------------------------------------------------
+# Notes - to remove this part when run for v13
+# Jo Lynne found the following status for 2/3 in the data warehouse:
+# C4106355-Tumor = Recurrence, C4113612-Tumor = Initial CNS Tumor, 7316UP-3705 = Not Reported
+# manually annotate them here and these will be updated in the v13 - then discard that part of the code
+#---------------------------------------------------------------------------------------------------------
+# The file contains duplicates because some samples might have: 
+# (1) both WGS and WXS, and/or (2) multiple biospecimen samples. 
+# We will address these issues by creating a unique identifier per each sample and assay (match_id_DNA).
+
+# make this code reproducible
+set.seed(2023)
+
+# Read in histologies file and filter for the pbta cohort
+pbta <- readr::read_tsv(v12_file, guess_max = 100000, show_col_types = FALSE) %>%
+  filter(cohort == "PBTA",
+         !is.na(pathology_diagnosis), # to remove matched normal samples
+         !composition %in% c("Derived Cell Line", "PDX"),
+         !experimental_strategy %in% c("Methylation", "Targeted Sequencing")) %>%
+  mutate(tumor_descriptor = case_when(grepl("C4106355-Tumor", sample_id) ~ "Recurrence",
+                                      grepl("C4113612-Tumor", sample_id) ~ "Diagnosis",
+                                      grepl("Primary Tumor|Initial CNS Tumor", tumor_descriptor) ~ "Diagnosis",
+                                      grepl("Progressive Disease Post-Mortem", tumor_descriptor) ~ "Deceased",
+                                      TRUE ~ tumor_descriptor),
+         match_id = paste(tumor_descriptor, sample_id, sep = "_"),
+         DNA = case_when(grepl("WGS|WXS", experimental_strategy) ~ "DNA",
+                         grepl("RNA-Seq", experimental_strategy) ~ "no"),
+         match_id_DNA = paste(Kids_First_Participant_ID, tumor_descriptor, DNA, sep = "_")) %>%
+  distinct(match_id_DNA, .keep_all = TRUE) # filter out
+
+print(pbta %>% count(tumor_descriptor))
```

Comment:
this is producing an error for me. alternatively you can try `print (count(pbta$tumor_descriptor_final))`

Reply:
This code should be for the `tumor_descriptor`, there is no `tumor_descriptor_final`.
I tried it anyway, but it produces an error for me (see below). Can you please confirm that you run this module via Docker?

> print (count(pbta$tumor_descriptor))
Error in UseMethod("count") : 
  no applicable method for 'count' applied to an object of class "character"